src/Compiler/AbstractIL/ilreflect.fs

```diff
@@ -913,20 +913,12 @@ let queryableTypeGetMethodBySearch cenv emEnv parentT (mref: ILMethodRef) =
             match a with
             | None -> true
             | Some a ->
-                if
-                    // obvious case
-                    p.IsAssignableFrom a
-                then
-                    true
-                elif
-                    p.IsGenericType
-                    && a.IsGenericType
-                    // non obvious due to contravariance: Action<T> where T: IFoo accepts Action<FooImpl> (for FooImpl: IFoo)
-                    && p.GetGenericTypeDefinition().IsAssignableFrom(a.GetGenericTypeDefinition())
-                then
-                    true
-                else
-                    false
+                // obvious case
+                p.IsAssignableFrom a
+                || p.IsGenericType
+                   && a.IsGenericType
+                   // non obvious due to contravariance: Action<T> where T: IFoo accepts Action<FooImpl> (for FooImpl: IFoo)
+                   && p.GetGenericTypeDefinition().IsAssignableFrom(a.GetGenericTypeDefinition())
```

Comment:
I find the more verbose case much easier to understand, can we please have it instead?